ipam/allocator.go

```diff
@@ -397,21 +420,23 @@ func (a *Allocator) getPredefinedPool(as string, ipV6 bool) (*net.IPNet, error)
 		return nil, err
 	}
 
-	for _, nw := range a.getPredefineds(as) {
+	predefined := a.getPredefineds(as)
+
+	aSpace.Lock()
+	for i, nw := range predefined {
 		if v != getAddressVersion(nw.IP) {
 			continue
 		}
-		aSpace.Lock()
 		if _, ok := aSpace.subnets[SubnetKey{AddressSpace: as, Subnet: nw.String()}]; ok {
-			aSpace.Unlock()
 			continue
 		}
 		if !aSpace.contains(as, nw) {
```

Comment:
This check not sure what is trying to achieve, I would expect that the network coming from the predefined pool is always non overlapping. Any idea of the real purpose for this?

Reply:
hrm....  Not sure...   Maybe there was some concern when doing the user-defined pools that one might have declared overlapping pools?